backends/shapeinference/shapeinference.go

```diff
@@ -319,16 +319,25 @@ func UnaryOp(opType backends.OpType, operand shapes.Shape) (output shapes.Shape,
 
 // WhereOp returns the shape resulting from the Where operation.
 //
-// Shape constraints for the operation:
+// Shape constraints for the operation (following StableHLO select specification):
 //
-//  1. The onTrue and onFalse must have the exact same shape, or one can be a scalar.
-//  2. The condition must either be a scalar or match the shape of onTrue or onFalse, except for the DType that
+//  1. The onTrue and onFalse must have the exact same dtype.
+//  2. The onTrue and onFalse must have the exact same shape, or one can be a scalar.
+//  3. The condition must either be a scalar or match the shape of onTrue or onFalse, except for the DType that
 //     must be Bool.
+//
+// Note: If you need to select between values of different dtypes, use ConvertDType to convert them
+// to a common dtype before calling Where.
 func WhereOp(condition, onTrue, onFalse shapes.Shape) (output shapes.Shape, err error) {
 	if condition.DType != dtypes.Bool {
 		err = errors.Errorf("condition for Where() must be a boolean, got %s instead", condition)
 		return
 	}
+	if onTrue.DType != onFalse.DType {
+		err = errors.Errorf("Where() requires onTrue and onFalse to have matching dtypes (per StableHLO spec): got onTrue=%s, onFalse=%s. Use ConvertDType to convert to a common dtype first",
```

Comment:
I was going to suggest to remove the "StableHLO" reference: the backend api (`backends.Builder.Where`) is the source of truth. It happens that the backend API for now follows the StableHLO one, but it could diverge in the future.

So better to document things in terms of the backend api specification. And if needed, updated the documentation there, it will be the source of truth so to say.

(Also, someone using `simplego` shouldnt' need to know what "StableHLO' is 😄 )
Since